main.go

```diff
@@ -0,0 +1,82 @@
+// http://127.0.0.1:8080/api/weather?q='your city'
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/joho/godotenv"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+)
+
+const (
+	weatherAPI = "http://api.openweathermap.org/data/2.5/weather"
+	units      = "metric"
+)
+
+type WeatherResponse struct {
+	Name string `json:"name"`
+	Main struct {
+		Temp float64 `json:"temp"`
+	} `json:"main"`
+}
+
+func main() {
+	http.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+		city := r.URL.Query().Get("q")
```

Comment:
I suggest using a clearer name for query parameter

Reply:
fixed

Reply:
fixed